server.go

```diff
@@ -1500,6 +1506,13 @@ func (s *Server) ReadSeries(database, retentionPolicy, name string, tags map[str
 // Returns a resultset for each statement in the query.
 // Stops on first execution error that occurs.
 func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User) Results {
+	// Authorize user to execute the query.
+	if err := Authorize(user, q, database); err != nil {
```

Comment:
OK, thanks.

This is not a huge point, as I expect these types of failures to be rare (1 statement of a set failing to execute due to an auth failure) but I still think we're not applying our API principles consistently -- currently it is assuming all statements parse, each statement is executed unless a failure is encountered at which stage processing stops but statements which did execute are fully executed. Unless we're adding a new rule to the API -- if any statement fails due to an auth failure, then no statement is executed. That's OK, I just want to be clear because obviously this wasn't specc'ed in the API doc, and you had to choose some policy.